### PR TITLE
DEV-16902 【0】 Use webmock in RSpec for this PR

### PR DESCRIPTION
**Pull request by levanmanh**
Template commit
## Title
DEV-16902 【0】 Use webmock in RSpec for this PR

## Ticket
- https://pixta.backlog.jp/view/DEV-16902

## Background
- Use webmock

## Release Date
- 20/10/2017

## Test
* [x] CI build pass
* [x] test by hand on staging env

### CODE REVIEW

File: spec/iamport_spec.rb [hunk 1]
```diff
@@ -31,28 +31,35 @@
...
-      response = {
-        "response" => {
-          "access_token" => "NEW_TOKEN",
-        },
-      }
+    let(:expected_url) { "#{IAMPORT_HOST}/users/getToken" }
 
-      expect(HTTParty).to receive(:post).with(expected_url, expected_params).and_return(response)
+    before do
+      stub_request(:post, expected_url)
```
Comment (haanhduclinh) [inline spec/iamport_spec.rb]: I see there are 6 places on this file we need replace
- https://github.com/pixta-dev/iamport-rest-client-ruby/blob/b6ecaeddfd6d76002e13f6b3ed981584b6dcdf6e/spec/iamport_spec.rb#L146
- https://github.com/pixta-dev/iamport-rest-client-ruby/blob/b6ecaeddfd6d76002e13f6b3ed981584b6dcdf6e/spec/iamport_spec.rb#L177
- https://github.com/pixta-dev/iamport-rest-client-ruby/blob/b6ecaeddfd6d76002e13f6b3ed981584b6dcdf6e/spec/iamport_spec.rb#L281
- https://github.com/pixta-dev/iamport-rest-client-ruby/blob/b6ecaeddfd6d76002e13f6b3ed981584b6dcdf6e/spec/iamport_spec.rb#L303
- https://github.com/pixta-dev/iamport-rest-client-ruby/blob/b6ecaeddfd6d76002e13f6b3ed981584b6dcdf6e/spec/iamport_spec.rb#L341
- https://github.com/pixta-dev/iamport-rest-client-ruby/blob/b6ecaeddfd6d76002e13f6b3ed981584b6dcdf6e/spec/iamport_spec.rb#L401
Please check it again

Comment (levanmanh) [reply spec/iamport_spec.rb]: :+1:

**Comment by haanhduclinh** [comment]
LGTM!

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 $LOAD_PATH.unshift File.expand_path("../../lib", __FILE__)
 require "iamport"
+require "webmock/rspec"
```
Comment (yasaichi) [inline spec/spec_helper.rb]: [IMO]
Maybe webmock doesn't allow any external requests as default (could you please investigate the behavior?), so you'll have to add the following line:

```ruby
WebMock.allow_net_connect!
```

File: spec/iamport_spec.rb [hunk 3]
```diff
@@ -31,28 +31,35 @@
...
   describe ".token" do
-    it "generates and returns new token" do
-      expected_url = "#{IAMPORT_HOST}/users/getToken"
-      expected_params = {
-        body: {
-          imp_key: api_key,
-          imp_secret: api_secret,
-        },
-      }
+    let(:expected_url) { "#{IAMPORT_HOST}/users/getToken" }
```
Comment (yasaichi) [inline spec/iamport_spec.rb]: [IMO]
It is referred in the before block, so you should move it into the block:

```ruby
before do
  expected_url = "#{IAMPORT_HOST}/users/getToken"
  stub_request(:post, expected_url).to ...
end
```

or

```ruby
before do
  stub_request(:post,  "#{IAMPORT_HOST}/users/getToken").to ...
end

```

**Comment by yasaichi** [comment]
@levanmanh @haanhduclinh 
I understand we forked the repo for some reasons, these problems've been already solved?
If not, we should create a pull request for it, what do you think?

File: spec/spec_helper.rb (hunk 2 above)
Comment (levanmanh) [reply spec/spec_helper.rb]: I think it's unnecessary, I checked and all passed

Comment (haanhduclinh) [reply spec/spec_helper.rb]: @yasaichi 
I see. I checked on Offical PR. Although, They have PR for `subscribe_customer` however they still lack some payments method `onetime` and `payment_again`. Ok. I will create pull request for this.

**Pull request merged**